src/core/Board.ts

```diff
@@ -42,13 +42,23 @@ class Board {
 
   movePiece(from: Position, to: Position): void {
     const piece = this.getPieceFromField(from);
+    const possibleCapturedPiece = this.getPieceFromField(to);
+
+    /* Removing captured piece from pieces */
+    if(possibleCapturedPiece) {
+      const indexToRemove: number = this.pieces.findIndex((el) => el === possibleCapturedPiece);
+      this.pieces.splice(indexToRemove, 1);
```

Comment:
It is good practice to check if index is greater than -1 before using it in splice (-1 returned in case item has not been found)